misc/json-to-record-converter/src/main/java/io/ballerina/jsonmapper/JsonToRecordMapper.java

```diff
@@ -392,11 +393,13 @@ private static void updateRecordFields(JsonObject jsonObject, Map<String, JsonEl
                 } else {
                     Token questionMarkToken = AbstractNodeFactory.createToken(SyntaxKind.QUESTION_MARK_TOKEN);
                     if (node1.kind().equals(SyntaxKind.ANYDATA_KEYWORD)) {
-                        nonAnyDataNode =
-                                NodeParser.parseTypeDescriptor(node2.toSourceCode() + questionMarkToken.text());
+                        nonAnyDataNode = NodeParser.parseTypeDescriptor(node2.toSourceCode());
+                        optionalFieldName = AbstractNodeFactory.createIdentifierToken(entry.getKey() +
+                                questionMarkToken.text());
                     } else if (node2.kind().equals(SyntaxKind.ANYDATA_KEYWORD)) {
-                        nonAnyDataNode =
-                                NodeParser.parseTypeDescriptor(node1.toSourceCode() + questionMarkToken.text());
+                        nonAnyDataNode = NodeParser.parseTypeDescriptor(node1.toSourceCode());
+                        optionalFieldName = AbstractNodeFactory.createIdentifierToken(entry.getKey() +
+                                questionMarkToken.text());
                     }
                 }
```

Comment:
We can rewrite this logic as below given that node.kind() is extracted to a variable.

```java
else if (node1Kind.equals(SyntaxKind.ANYDATA_KEYWORD) ||
                        node2Kind.equals(SyntaxKind.ANYDATA_KEYWORD)) {
    nonAnyDataNode = NodeParser.parseTypeDescriptor(node1Kind.equals(SyntaxKind.ANYDATA_KEYWORD) ?
            node2.toSourceCode() : node1.toSourceCode());
    Token questionMarkToken = AbstractNodeFactory.createToken(SyntaxKind.QUESTION_MARK_TOKEN);
    optionalFieldName = AbstractNodeFactory.createIdentifierToken(entry.getKey() +
            questionMarkToken.text());
}
```

Reply:
Let's not focus on these improvements for the moment as this PR is urgent